File: test_case_1/Pages/HomePage.cs
using Microsoft.Playwright;

namespace test_case_1.Pages
{
    public class HomePage : BasePage
    {
        public HomePage(IPage page) : base(page)
        {
        }

        private ILocator SignupLoginButton => _page.Locator("a[href='/login']");
        private ILocator HomePageContent => _page.Locator(".features_items");

        public Task NavigateToHomePageAsync(string url)
        {
            return _page.GotoAsync(url);
        }

        public Task ClickSignupLoginButtonAsync()
        {
            return ClickElementAsync(SignupLoginButton);
        }

        public Task<bool> IsHomePageContentVisibleAsync()
        {
            return IsElementVisibleAsync(HomePageContent);
        }
    }
}

File: test_case_3/pages/home_page.py
from pages.base_page import BasePage
from playwright.sync_api import expect

class HomePage(BasePage):

    SIGN_UP_LOGIN_LINK: str = "a[href='/login']"
    BODY: str = "body"
    TITLE: str = "Automation Exercise"
    
    def navigate_to_home_async(self, url: str):
        self.navigate_to_async(url)
                
    def verify_home_page_async(self):
        expect(self.page).to_have_title(self.TITLE)
        expect(self.page.locator(self.BODY)).to_be_visible()
    
    def click_signup_login_async(self):
        self.click_element_async(self.page.locator(self.SIGN_UP_LOGIN_LINK))